rio/srcrobot/robot_container.py

```diff
@@ -199,23 +199,23 @@ def autoModeShot(self, autoShot: Constants.NextShot) -> Command:
         Used during automode commands to shoot any Constants.NextShot.
         """
         self.m_robotState.m_gameState.setNextShot(autoShot)
-        armTimeoutSec = (autoShot.m_armAngle / 45.0) + 1.0
+        shotTimeoutSec = (autoShot.m_armAngle / 45.0) + 1.0
         return SequentialCommandGroup(
             InstantCommand(
                 lambda: self.s_Shooter.setShooterVelocity(autoShot.m_shooterVelocity),
                 self.s_Shooter,
             ).alongWith(
-                self.s_Arm.servoArmToTarget(autoShot.m_armAngle).withTimeout(
-                    armTimeoutSec
+                self.s_Arm.servoArmToTarget(autoShot.m_armAngle),
```

Comment:
After testing this in the sim, the shot may come out quickly but the command never ends, so the auton never moves on. Since the servoArmToTarget is in parallel with everything else, the fact that it doesn't end is causing the whole group to run forever. I think this needs either a deadline group (so the servoArmCommand gets canceled when the shot command finishes) or changing the servoArmToTarget command to be a runOnce (though deadline is probably better).

Reply:
And when the shot is set (outside the command) is very wrong too. Retry coming soon.

Reply:
I like this next try much better. The setting of the next shot is now first inside the command, the parallel actions are in a deadline group with the deadline being the arm and shooter ready. I left a timeout (increased by 0.5 sec) on the parallel command group as a whole. I think this is needed in case the arm or shooter does not get on target. The timeout could just be on the deadline command but I think it reads a little better this way.

Reply:
Oops, that is not quite right either. Another change coming soon.

Reply:
Move shoot inside the parallel group to keep the arm under servo control during the shot. Put the timeout back to what it was before.